Process input handler refactor

zrock-dev opened the pull request:
# Task-207 Implementation of Input intermediate

https://tree.taiga.io/project/joseluis-teran-coffeetime/task/207

## Description
Implementation of : https://github.com/Pending-Name-21/arquitecture/pull/20

## Type of Change

Put an `x` in all the boxes that apply:

- [ ] ✨ New Feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug Fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking Change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code Refactor
- [ ] ✅ Build Configuration Change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

## Pull Request Checklist

Put an `x` in all the boxes that apply:

- [x] My commit messages are detailed.
- [x] My code follows the code style of this project.
- [ ] No existing features have been broken without good reason.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests pass.

zrock-dev committed:
chore: updaded generated CoffeeTime

based on update of input-events.fbs schema

zrock-dev committed:
refactor: process input handler

refactored implementation because input-events.fbs schema.

- KeyboardEventPublisher has been abstracted to AEventManager which
  notifies to IEventSubscriber and is generic so T would be a Event like
  Keyboard or mouse.

- Modified Receiver to unserialize Event, and feed a IBuffer instead of
  a direct class.

codeFactory12 commented on bridge/src/main/java/com/bridge/processinputhandler/AEventManager.java:
```diff
@@ -0,0 +1,17 @@
+package com.bridge.processinputhandler;
+
```
src/main/java/com/bridge/processinputhandler/AEventManager.java
          @@ -1,7 +1,6 @@
           package·com.bridge.processinputhandler;
           
           import·com.bridge.ipc.IEventBuffer;
          -
           import·java.util.ArrayList;
           import·java.util.List;
           import·java.util.concurrent.ConcurrentLinkedQueue;
Doesnt acomlish with format, review with ./gradlew build

Run './gradlew :bridge:spotlessApply'

codeFactory12 replied:
There are more archieves that needs a format fix such us:
src/test/java/com/bridge/ipc/ReceiverTest.java
src/test/java/com/bridge/processinputhandler/AEventManagerTest.java
src/test/java/com/bridge/processinputhandler/InputVerifierTest.java
src/test/java/com/bridge/GameTest.java
src/test/java/com/bridge/EventGenerator.java
src/main/java/com/bridge/processinputhandler/IPublisher.java
can be fixed using : **./gradlew :bridge:spotlessApply**

codeFactory12 commented on bridge/src/test/java/com/bridge/GameTest.java:
```diff
@@ -1,18 +1,22 @@
 import com.bridge.initializerhandler.GameInitializer;
 import com.bridge.processinputhandler.InputVerifier;
+import com.bridge.processinputhandler.KeyboardEventManager;
+import com.bridge.processinputhandler.MouseEventManager;
```
There is no MouseEventManager uploaded

codeFactory12 commented on bridge/src/main/java/com/bridge/processinputhandler/KeyboardEventManager.java:
```diff
@@ -26,26 +24,23 @@
     /**
      * Constructs a KeyboardEventPublisher with the specified event buffer.
      *
      * @param buffer the concurrent linked queue used to store keyboard events
```
Cannot resolve symbol 'buffer' on line 32

codeFactory12 reviewed (changes requested):
There are errors in tests clases that are not uploeaded sich us 
test/java/com/bidge
ipc/SocketServerTest
Expected 0 arguments but found 1 on line 13
processinputhandler/EventTypeTest
Cannot resolve symbol 'EventType' on line 12
GameWithExceptionsTest
Cannot resolve symbol 'Utils'

zrock-dev committed:
fix: fixed failing tests

There was an error with how the test was written, fixed by updating the
event name.

zrock-dev committed:
doc: updated processinputhandler javadocs

codeFactory12 reviewed (approved):
Nice joob
